pkg/execution/singleton/singleton.go

```diff
@@ -17,7 +16,7 @@ var (
 )
 
 type Singleton interface {
-	Singleton(ctx context.Context, key string, c inngest.Singleton) (bool, error)
+	HandleSingleton(ctx context.Context, key string, c inngest.Singleton) (*string, error)
```

Comment:
if you're always expecting a runID here, why not return a `*ulid.ULID` instead of `*string`?